### PR TITLE
Feat/Utils

### PR DESCRIPTION
**Pull request by Debarshi95**
PR to add auth pages and other Util pages and components

Preview Link: https://627ec378dac740146b177b12--debarshib-wetube.netlify.app/
Credentials:
Email: devb13@test.com
Password: 1234567

### CODE REVIEW

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    try {
+      const res = await updatePlaylist({
+        variables: {
+          videoId,
+          playlistId,
+        },
+      });
+      const { success = false, message = 'Some error occurred' } = res.data?.updatePlaylist || {};
+      if (success) {
+        refectPlaylist();
```
Comment (SinchanaMS) [inline src/components/Modal/Modal.js:66]: There's a typo here, please check.

File: src/components/PlayerCard/PlayerCard.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  };
+
+  return (
+    <section className={cn('PlayerCard__root', className)}>
+      {children}
+      <div className="PlayerCard__contentWrapper">
+        <div className="d-flex content-between">
+          <div
+            role="button"
+            aria-hidden
```
Comment (SinchanaMS) [inline src/components/PlayerCard/PlayerCard.js:53]: I wasn't aware of using the button role. On reading further, I found this -
"Adding `role="button"` tells assistive technology that the element is a button but provides no button functionality. Use `button` or `input` with `type="button"` instead."

Just curious, any specific reason to use `div` here?

**Review by SinchanaMS — APPROVED** [review]
Lots of new things that I'm not aware of, have to read about them to comment.
In the UI I found that once we add a video to the playlist, we can't close the playlist modal. Please check on it later.

File: src/components/PlayerCard/PlayerCard.js (hunk 2 above)
Comment (Debarshi95) [reply src/components/PlayerCard/PlayerCard.js:53]: Thanks for pointing. Just to clear, it's allowed to  use div as button with role if any listener is attached to it. In my case, I have attached an onClick listener.

File: src/components/PlayerCard/PlayerCard.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                  role="button"
+                  aria-hidden
+                  onClick={(e) => {
+                    e.preventDefault();
+                    e.stopPropagation();
+                    prop?.onClick({ user, video });
+                  }}
+                >
+                  {prop?.icon}
+                </div>
```
Comment (saurabhchirde) [inline src/components/PlayerCard/PlayerCard.js:80]: JSX would be leaner and easy to read if you bring this mapping outside JSX

File: src/components/Modal/Modal.css [hunk 4]
```diff
@@ -0,0 +1,53 @@
+.Modal__root {
+  height: 100vh;
+  width: 100%;
+  position: fixed;
+  top: 0;
+  background-color: rgb(0, 0, 0);
```
Comment (ashishkatiyar2404) [inline src/components/Modal/Modal.css:6]: delete this one as u already used in next line

File: src/components/Modal/Modal.css [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+.Modal__form > input {
+  padding: 10px 12px;
+  width: 100%;
+  margin-bottom: 1rem;
+  border-radius: 4px;
+  font-size: 1rem;
+  outline: none;
+  border: none;
+  background-color: var(--color-dark);
+  color: #f7f7f7;
```
Comment (ashishkatiyar2404) [inline src/components/Modal/Modal.css:28]: You can use variables here also, as u already used the above code. 
If the same color is not repeated thn its okay

**Review by saurabhchirde — APPROVED** [review]
Couldn't find any issue, looks good to merge.

**Review by ashishkatiyar2404 — APPROVED** [review]
UI - Your UI looks majestic, your loading logo is 💥 just one suggestion add a margin to the bottom so it has space between the content and window.

curious question:- why did u capitalize the first alphabet of the class's name? Is that rule or ?

Other looks good to me, you can merge now.

File: src/components/PlayerCard/PlayerCard.js (hunk 3 above)
Comment (Debarshi95) [reply src/components/PlayerCard/PlayerCard.js:80]: cardActionsProps can be both array or simple object. The prop values are meant to present inside  the JSX, also based on this the handlers are attached and the callbacks are triggered. Hence, the mapping is itself inside the JSX.

**Comment by Debarshi95** [comment]
> UI - Your UI looks majestic, your loading logo is boom just one suggestion add a margin to the bottom so it has space between the content and window.
> 
> curious question:- why did u capitalize the first alphabet of the class's name? Is that rule or ?
> 
> Other looks good to me, you can merge now.

Thanks for the comment and appreciation. Tried to follow a common naming convention, there is a naming convention called BEM. Hence, they're capitalized

**Pull request merged**